scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/DTLSSession.java

```diff
@@ -243,6 +262,37 @@ void setSessionIdentifier(SessionId sessionIdentifier) {
 		this.sessionIdentifier = sessionIdentifier;
 	}
 
+	/**
+	 * Gets the (virtual) host name at the server that this session
+	 * has been established for.
+	 * 
+	 * @return the host name or {@code null} if this session has not
+	 *         been established for a virtual host.
+	 */
+	public String getVirtualHost() {
+		return virtualHost;
+	}
+
+	void setVirtualHost(String hostname) {
+		this.virtualHost = hostname;
+	}
+
+	/**
+	 * Checks whether the peer (the server) supports
+	 * the Server Name Indication extension.
+	 * 
+	 * @return {@code true} if the server has included
+	 *         an empty SNI extension in its SERVER_HELLO
+	 *         message during handshake.
+	 */
+	public boolean isSniSupported() {
+		return peerSupportsSni;
+	}
+
+	void setSniSupported(boolean flag) {
+		this.peerSupportsSni = flag;
+	}
+
 	public DtlsEndpointContext getConnectionWriteContext() {
 		return new DtlsEndpointContext(peer, peerIdentity, sessionIdentifier.toString(),
```

Comment:
Please add the SNI/virtualHost to the context.

Reply:
That will be in the next step ...

Reply:
OK, then please also add this to the read-context.